tests/admin_widgets/tests.py

```diff
@@ -1813,12 +1834,28 @@ def _submit_and_wait():
             ),
         )
         # "Currently" with "Clear" checkbox and "Change" still shown.
-        cover_field_row = self.selenium.find_element(By.CSS_SELECTOR, ".field-photo")
-        self.assertIn("Currently", cover_field_row.text)
-        self.assertIn("Change", cover_field_row.text)
-        # "Clear" box works.
-        self.selenium.find_element(By.ID, clear_checkbox_id).click()
-        _submit_and_wait()
-        student.refresh_from_db()
-        self.assertEqual(student.name, "Joe Doe")
-        self.assertEqual(student.photo.name, "")
+        photo_field_row = self.selenium.find_element(By.CSS_SELECTOR, ".field-photo")
+        self.assertIn("Currently", photo_field_row.text)
+        self.assertIn("Change", photo_field_row.text)
+
+    def test_clearablefileinput_widget_persist_clear_checkbox(self):
+        from selenium.webdriver.common.by import By
+
+        self._run_happy_image_upload_path()
+        # "Clear" is not checked by default.
+        self.assertFalse(
+            self.selenium.find_element(By.ID, self.CLEAR_CHECKBOX_ID).is_selected()
+        )
+        # "Clear" was checked, but a validation error is raised.
+        name_input = self.selenium.find_element(By.ID, self.NAME_INPUT_ID)
+        name_input.clear()
+        self.selenium.find_element(By.ID, self.CLEAR_CHECKBOX_ID).click()
+        self._submit_and_wait()
+        self.assertEqual(
+            self.selenium.find_element(By.CSS_SELECTOR, ".errorlist li").text,
+            "This field is required.",
+        )
+        # "Clear" persists checked.
+        self.assertTrue(
+            self.selenium.find_element(By.ID, self.CLEAR_CHECKBOX_ID).is_selected()
+        )
```

Comment:
Some of existing tests in `ClearableFileInputTest` already fail, so maybe we don't need new selenium tests :thinking:

Reply:
When I was debugging it (from the browser) and I edited the `checked` attribute manually sometimes it didn't take the change 🤔 , so running from the browser gives me more confidence than just assert the html during the html render.